src/firmware/app/control/python_controller/difference_equation.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+"""
+This file contains the implementation of the DifferenceEquation class which generates a difference equation based on an input discrete transfer function.
+
+"""
+
+import numpy as np
+import control as ct
+from control.matlab import *
+import matplotlib.pyplot as plt
+import collections
+import _collections_abc
+import os
+
+
+class DifferenceEquation:
+    def __init__(
+        self, tf_numerator_coefficients: list, tf_denominator_coefficients: list
+    ):
+
+        """ Creates a difference equation based on the input discrete domain transfer function
+
+        :list tf_numerator_coefficients: The coefficients of the numerator of the discrete transfer function
```

Comment:
seems like the type (`list`) is duplicated in the doc comment and in the function signature. I think we should be good to remove it from the doc comment?

Reply:
This is the documentation style that @MathewMacDougall started doing with the python code

Reply:
oh did I? I agree with gareth lets try just keep the type annotation in the function signature, since that's what actually matters when running it (and comments likely won't stay up to date)

Reply:
Removed the extra type specifications